crates/red_knot_python_semantic/resources/mdtest/type_of/any.md

```diff
@@ -0,0 +1,43 @@
+# type[Any]
+
+## Simple
+
+```py
+def f(x: type[Any]):
+    reveal_type(x)  # revealed: type[Any]
+    reveal_type(x.__repr__)  # revealed: Any
+
+class A: ...
+
+f(object)
+f(type)
+f(A)
```

Comment:
Note that we do not have call-signature checking, so this doesn't test anything yet. If you want to check for assignability, you would have to do something like
```suggestion
x: type[Any] = object
x: type[Any] = type
x: type[Any] = A
```
For some reason, those assignments currently seem to fail(?).

Reply:
Oh, great catch! Yeah, if those assignments are failing, we should definitely understand why.

Reply:
I think assignability falls back to subtyping, and `type[Any]` does not participate in subtyping. We probably need something like this special case in `Type::is_assignable_to`:
```rs
(Type::ClassLiteral(_) | Type::SubclassOf(_), Type::SubclassOf(target))
    if target.base == ClassBase::Any =>
{
    true
}
```

It might make sense to add a new `Ty::SubclassOfAny` (and `Ty::SubclassOf(…)`) variant to our testing-enum `Ty` and to write a few explicit unit tests for this type?

For example, I think we currently treat `type[Any]` as fully static.